File: src/002. Even Fibonacci numbers/002.lua
local a1, a2 = 1, 2
local sum = 0

while a1 < 4000000 do
  a1, a2 = a2, a1 + a2
  if a1 % 2 == 0 then
    sum = sum + a1
  end
end

print(sum)

File: src/002. Even Fibonacci numbers/002.go
package main

import "fmt"

func main () {
	f := make(map[int]int)
	f[1] = 1
	f[2] = 1

	for i := 3; ; i++ {
		f[i] = f[i - 2] + f[i - 1]
		if f[i] > 4000000 {
			break
		}
	}

	sum := 0
	for _, v := range f {
		if v % 2 == 0 {
			sum += v
		}
	}

	fmt.Println(sum)
}
